keyboard/src/data/keyboards.rb

```diff
@@ -266,6 +266,31 @@ def self.all_keyboards
     ]
   end
 
+  # Some keyboards are present in new openSUSE releases but not in older SLE
+  # @see all_keyboards
+  def self.optional_keyboards
+    # memoize this
+    return @optional_keyboards unless @optional_keyboards.nil?
+
+    @optional_keyboards = []
+
+    # The afnor layout was added to xkeyboard-config in 2019-06
+    # but SLE15-SP4 only has 2.23 released in 2018
+    afnor_test = lambda do
+      kmm = File.read("/usr/share/systemd/kbd-model-map") rescue ""
+      kmm.match?("^fr-afnor")
+    end
+    afnor = {
+      "description" => _("French (AFNOR)"),
+      "alias" => "french-afnor",
+      "code" => "fr-afnor",
+      "suggested_for_lang" => ["br_FR", "fr", "fr_BE"]
+    }
+    @optional_keyboards.push(afnor) if afnor_test.call
+
+    @optional_keyboards
```

Comment:
Maybe you can use `Yast::Execute` here

```suggestion
    # TODO: put below constants in a proper place
    GREP_CMD = "/usr/bin/grep".freeze
    KDB_MODEL_MAP = "/usr/share/systemd/kbd-model-map".freeze
    
    if Yast::Execute.on_target!(GREP_CMD, "^fr-afnor", KDB_MODEL_MAP)
      # The afnor layout was added to xkeyboard-config in 2019-06
      # but SLE15-SP4 only has 2.23 released in 2018
      @optional_keyboards.push(afnor_keyboard_layout)
    end
  rescue Cheetah::ExecutionFailed => e
    # TODO: log something?
  ensure
    @optional_keyboards
  end
  
  def afnor_keyboard_layout
    {
        "description" => _("French (AFNOR)"),
        "alias" => "french-afnor",
        "code" => "fr-afnor",
        "suggested_for_lang" => ["br_FR", "fr", "fr_BE"]
    }
  end
  
```

Reply:
:thinking: :thinking: 

Still thinking that `Yast::Execute` could be an option, but not with the above approach since once it fails it will not add _the rest_ of optional keyboards when adding more.

Reply:
Well, "executing grep" is a nice conceptual shortcut that we're familiar with from the shell, and I 've thought of it too, but decided to implement it in Ruby directly. It is simple and faster.